src/table.js

```diff
@@ -538,15 +543,105 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type) {
+  const stringValue = typeof value === "string" ? value.trim() : value;
+  switch (type) {
+    case "string":
+      return typeof value === "string"
+        ? stringValue
+        : value || value === 0
+        ? value.toString()
+        : null;
+    case "boolean":
+      return value === true || stringValue === "true"
+        ? true
+        : value === false || stringValue === "false"
+        ? false
+        : null;
+    case "integer":
+      return value === 0
+        ? value
+        : !value || isNaN(parseInt(value))
+        ? NaN
+        : parseInt(value);
+    case "bigint":
+      return typeof value === "bigint"
+        ? value
+        : value === 0 || value === true || value === false
+        ? // eslint-disable-next-line no-undef
+          BigInt(value)
+        : !value || isNaN(value) || !Number.isInteger(+value)
+        ? NaN
+        : // eslint-disable-next-line no-undef
+          BigInt(value);
+    case "number": {
+      return value === 0
+        ? value
+        : !value || isNaN(value)
+        ? NaN
+        : Number(value);
+    }
+    case "date": {
+      if (value instanceof Date) return value;
+      if (typeof value === "string") {
+        let match;
+        if (
+          (match = stringValue.match(
+            /^([-+]\d{2})?\d{4}(-\d{2}(-\d{2})?)?(T\d{2}:\d{2}(:\d{2}(\.\d{3})?)?(Z|[-+]\d{2}:\d{2})?)?$/
+          ))
+        ) {
+          if (fixTz && !!match[4] && !match[7])
+            value = stringValue.replace(/-/g, "/").replace(/T/, " ");
+        }
+      }
+      return new Date(value);
+    }
+    case "array":
+      if (Array.isArray(value)) return value;
+      return [value];
+    case "object":
+      // this will return true for everything except null, undefined, strings,
+      // numbers, boolean, and symbols, so may yield unexpected results.
+      if (typeof value === "object") return value;
+      return {value: value};
+    case "buffer":
+    case "other":
```

Comment:
Same here, I don’t think these should be offered as target types for coercion.

Reply:
Agreed, I've updated the visible options for the user in this [draft PR](https://github.com/observablehq/observablehq/pu
<img width="186" alt="Screen Shot 2023-02-01 at 1 05 53 PM" src="https://user-images.githubusercontent.com/1373882/216126259-f15fbaad-1acd-4f10-b101-22a79f9eec28.png">
ll/10133).

Reply:
Thanks, @mkfreeman! Let's also remove "Integer" from that list.

Reply:
Happy to, but out of curiosity, is that because we don't expect users to need their data as integers, or because the coercion is tricky?

Reply:
I was agreeing with Mike B's comment above, that they can round values themselves in JavaScript or SQL, if that's what they want.

Reply:
Ah, missed that comment. Removing `interger` as a target!

Reply:
I think for these unusual types we should just `return value` here and have coercion do nothing. I don’t think we want to convert `false` or the empty string `""` to `null`, for example.